openmc/mgxs/mgxs.py

```diff
@@ -4618,3 +4649,415 @@ def get_pandas_dataframe(self, groups='all', nuclides='all',
+        is None unless the multi-group cross section has been computed.
+    num_subdomains : int
+        The number of subdomains is unity for 'material', 'cell' and 'universe'
+        domain types. This is equal to the number of cell instances
+        for 'distribcell' domain types (it is equal to unity prior to loading
+        tally data from a statepoint file).
+    num_nuclides : int
+        The number of nuclides for which the multi-group cross section is
+        being tracked. This is unity if the by_nuclide attribute is False.
+    nuclides : Iterable of str or 'sum'
+        The optional user-specified nuclides for which to compute cross
+        sections (e.g., 'U-238', 'O-16'). If by_nuclide is True but nuclides
+        are not specified by the user, all nuclides in the spatial domain
+        are included. This attribute is 'sum' if by_nuclide is false.
+    sparse : bool
+        Whether or not the MGXS' tallies use SciPy's LIL sparse matrix format
+        for compressed data storage
+    loaded_sp : bool
+        Whether or not a statepoint file has been loaded with tally data
+    derived : bool
+        Whether or not the MGXS is merged from one or more other MGXS
+    hdf5_key : str
+        The key used to index multi-group cross sections in an HDF5 data store
+
+    """
+
+    def __init__(self, domain=None, domain_type=None,
+                 groups=None, by_nuclide=False, name=''):
+        super(Velocity, self).__init__(domain, domain_type,
+                                              groups, by_nuclide, name)
+        self._rxn_type = 'velocity'
+
+    @property
+    def scores(self):
+        return ['inverse-velocity', 'flux']
+
+    @property
+    def rxn_rate_tally(self):
+        if self._rxn_rate_tally is None:
+            self._rxn_rate_tally = self.tallies['flux']
+            self._rxn_rate_tally.sparse = self.sparse
+        return self._rxn_rate_tally
+
+    @property
+    def xs_tally(self):
+
+        if self._xs_tally is None:
+            inverse_velocity = self.tallies['inverse-velocity']
+
+            # Compute the velocity
+            self._xs_tally = self.rxn_rate_tally / inverse_velocity
+            super(Velocity, self)._compute_xs()
+
+        return self._xs_tally
+
+    def get_units(self, xs_type='macro'):
+        r"""Returns the units of Velocity.
+
+        This method returns the units of a Velocity based on a desired xs_type.
+
+        Parameters
+        ----------
+        xs_type: {'macro', 'micro'}
+            Return the macro or micro cross section units.
+            Defaults to 'macro'.
+
+        Returns
+        -------
+        str
+            A string representing the units of the Velocity.
+
+        """
+
+        cv.check_value('xs_type', xs_type, ['macro', 'micro'])
+
+        if xs_type == 'macro':
+            return 'cm/second'
+        else:
+            raise ValueError('Unable to return the units of Velocity for ' +
```

Comment:
The `+` at the end is superfluous (Python concatenates whitespace-separated strings).